### PR TITLE
Updated response text

### PR DESCRIPTION
**Pull request by peterbrightwell**
Responses are now MUSTs rather than SHOULDs. Clarified that any request error means entire request is rejected. Required 4xx codes now in a table.

Merge after #40 and #41.

### CODE REVIEW

File: docs/4.0. Behaviour.md [hunk 1]
```diff
@@ -230,29 +230,30 @@ Each activation is allocated a unique identifier by the API. This identifier is
...
+An HTTP 202 response MUST be returned when a request for a scheduled activation is accepted, to indicate that while the request itself was acceptable it has not yet been acted upon by the sender. For scheduled activations `activation_time` SHOULD be the absolute TAI time the parameters will actually transition. For absolute scheduled activations SHOULD be the same as the requested time, but may differ if the Device is unable to schedule at the requested time.
 
-If an Output is modified by an already scheduled activation the API SHOULD respond with HTTP 423 `resource locked`. The API MUST reject the entire request, even if some of the routes in the failed request are not currently locked by another activation.
+In there is an error in the request, the API MUST reject the entire request with an appropriate 4xx response. Particular responses MUST be used in these cases:
```
Comment (garethsb) [inline docs/4.0. Behaviour.md]:
```suggestion
When there is an error in the request, the API MUST reject the entire request with an appropriate 4xx response. Particular responses MUST be used in the following cases:
```

(There are other cases, like completely invalid JSON.)

File: docs/4.0. Behaviour.md [hunk 2]
```diff
@@ -230,29 +230,30 @@ Each activation is allocated a unique identifier by the API. This identifier is
...
 
-Any attempt to set one of the two fields within a Output channel object to `null` without setting the other SHOULD result in an HTTP 400 response. In this error scenario the API MUST reject the activation request in full, even if some of the routes in the failed request would have been valid.
+| Error | HTTP Response |
+| --- | --- |
+| Output is modified by an already scheduled activation. | 423 (Locked) |
+| Routing request not permitted by Input or Output constraints | 400 (Bad Request). |
```
Comment (garethsb) [inline docs/4.0. Behaviour.md]:
```suggestion
| Routing request not permitted by Input or Output constraints. | 400 (Bad Request) |
```

File: docs/4.0. Behaviour.md (hunk 1 above)
Comment (garethsb) [inline docs/4.0. Behaviour.md]: Can more than one of these cases apply at the same time? Does that warrant any extra wording?

File: docs/4.0. Behaviour.md [hunk 3]
```diff
@@ -230,29 +230,30 @@ Each activation is allocated a unique identifier by the API. This identifier is
...
+For activations the `mode` parameter MUST return to `null` once the activation transaction with the client is completed (i.e the response to the PATCH has been sent by the server).
 
 #### Canceling Activations
 
 A client cancels an activation by submitting a DELETE request to the resource `/map/acivations/activation-id`. If the activation has already been completed the API SHOULD respond with an HTTP 404 error response, as the activation no longer exists.
 
 ### Timestamps
 
-All timestamps in this API MUST be string formatted timestamps (\<seconds\>:\<nanoseconds\>) indicating time (absolute or relative) for activation. For relative times, the time '0:0' MUST be considered to be the time the request is received by the API.
+Timestamps indicate the absolute or relative time for activation and MUST be string formatted as `<seconds>:<nanoseconds>`. Absolute timestamps MUST be based on TAI (which does not apply leap seconds) and must use SMPTE epoch (meaning 1970-01-01 00:00:00 TAI would be represented as `0:0`).
```
Comment (garethsb) [inline docs/4.0. Behaviour.md]: Bit confused why I'm seeing these timestamp related changes in this commit (https://github.com/AMWA-TV/nmos-audio-channel-mapping/pull/42/commits/06adf659c7e29d423d33d9aeaa3edab1d6c681de), when they appear to be in https://github.com/AMWA-TV/nmos-audio-channel-mapping/pull/41/commits/334110a1a54651fb468f1da770a14428425347fa. Just something to keep an eye on when we merge.

File: docs/4.0. Behaviour.md [hunk 4]
```diff
@@ -226,40 +230,64 @@ Each activation is allocated a unique identifier by the API. This identifier is
...
+| Error | HTTP Response |
+| --- | --- |
+| Output is modified by an already scheduled activation. | 423 (Locked) |
+| Routing request not permitted by Input or Output constraints | 400 (Bad Request). |
+| One field of Output channel object is `null` but other is not. | 400 (Bad Request) |
 
-In the event that the stage request attempts to perform a routing operation that is not permitted by the Input or Output routing constraints the API SHOULD respond with an HTTP 400 error response. The API SHALL reject the entire request, even if some of the routes in the failed request would have been valid.
 
-Note that for activations the `mode` parameter MUST return to null once the activation transaction with the client is completed (i.e the response to the PATCH has been sent by the server).
+For activations the `mode` parameter MUST return to `null` once the activation transaction with the client is completed (i.e the response to the PATCH has been sent by the server).
```
Comment (garethsb) [inline docs/4.0. Behaviour.md:245]: So far as I can see the API doesn't use PATCH requests on any endpoint, I assume this sentence should be referring to POST requests. But PATCH requests are also mentioned in the Overview and in the schemas, so I'll raise a separate issue.

Comment (peterbrightwell) [reply docs/4.0. Behaviour.md:245]: Moved to issue #43

File: docs/4.0. Behaviour.md (hunk 3 above)
Comment (peterbrightwell) [reply docs/4.0. Behaviour.md]: Will be resolved in merge.

File: docs/4.0. Behaviour.md (hunk 1 above)
Comment (peterbrightwell) [reply docs/4.0. Behaviour.md]: We think it's ok.

File: docs/4.0. Behaviour.md (hunk 2 above)
Comment (peterbrightwell) [reply docs/4.0. Behaviour.md]: Resolved (twice now :) )

File: docs/4.0. Behaviour.md (hunk 1 above)
Comment (peterbrightwell) [reply docs/4.0. Behaviour.md]: Resolved

**Pull request merged**